R/summary-insertions.R

```diff
@@ -0,0 +1,261 @@
+#'
+#' @examples
+#' \dontrun{
+#' pet_feelings
+#' #                   animals
+#' #feelings            cats dogs
+#' #  extremely happy      9    5
+#' #  somewhat happy      12   12
+#' #  neutral             12    7
+#' #  somewhat unhappy    10   10
+#' #  extremely unhappy   11   12
+#'
+#' # add a mean summary statistic to a CrunchCube
+#' addSummaryStat(pet_feelings, stat = "mean", var = "feelings")
+#' #                 animals
+#' #feelings                      cats             dogs
+#' #  extremely happy                9                5
+#' #   somewhat happy               12               12
+#' #          neutral               12                7
+#' # somewhat unhappy               10               10
+#' #extremely unhappy               11               12
+#' #             mean 4.90740740740741 4.34782608695652
+#'
+#' # we can also store the CrunchCube for use elsewhere
+#' pet_feelings <- addSummaryStat(pet_feelings, stat = "mean", var = "feelings")
+#' pet_feelings
+#' #                 animals
+#' #feelings                      cats             dogs
+#' #  extremely happy                9                5
+#' #   somewhat happy               12               12
+#' #          neutral               12                7
+#' # somewhat unhappy               10               10
+#' #extremely unhappy               11               12
+#' #             mean 4.90740740740741 4.34782608695652
+#'
+#' # `addSummaryStat` returns a CrunchCube that has had the summary statistic
+#' # added to it, so that you can still use the Crunch logic for multiple
+#' # response variables, missingness, etc.
+#' class(pet_feelings)
+#' #[1] "CrunchCube"
+#' #attr(,"package")
+#' #[1] "crunch"
+#'
+#' # cleanup transforms
+#' transforms(pet_feelings) <- NULL
+#' # add a median summary statistic to a CrunchCube
+#' pet_feelings <- addSummaryStat(pet_feelings, stat = "median", var = "feelings")
+#' pet_feelings
+#' #                 animals
+#' #feelings             cats    dogs
+#' #  extremely happy       9       5
+#' #   somewhat happy      12      12
+#' #          neutral      12       7
+#' # somewhat unhappy      10      10
+#' #extremely unhappy      11      12
+#' #           median       5       5
+#'
+#' # additionally, if you want a true matrix object from the CrunchCube, rather
+#' # than the CrunchCube object itself, `applyTransforms()` will return the
+#' # array with the summary statistics (just like subtotals and headings)
+#' pet_feelings_array <- applyTransforms(pet_feelings)
+#' pet_feelings_array
+#' #                 animals
+#' #feelings             cats    dogs
+#' #  extremely happy       9       5
+#' #   somewhat happy      12      12
+#' #          neutral      12       7
+#' # somewhat unhappy      10      10
+#' #extremely unhappy      11      12
+#' #           median       5       5
+#'
+#' # and we can see that this is a matrix, and no longer a CrunchCube
+#' class(pet_feelings_array)
+#' #[1] "matrix"
```

Comment:
why is this "matrix" and not "array" or "table"?

Reply:
As far as I can tell, this is baked into R: all 2-dimensional arrays have the `class` matrix.

```
> cat_x_mr <- loadCube(test_path("cubes/selected-crosstab-array-last.json"))
> class(as.array(cat_x_mr))
[1] "matrix"
> is.array(as.array(cat_x_mr))
[1] TRUE

> cat_x_mr_x_mr <- loadCube(test_path("cubes/cat-x-mr-x-mr.json"))
> class(as.array(cat_x_mr_x_mr))
[1] "array"
```

And even:
```
> class(array(data = 1:4, dim = c(2,2)))
[1] "matrix"
```

I've changed this example to use `is.array()` to hopefully be less confusing.